src/main/java/net/neoforged/neoforge/event/village/VillagerChangeProfessionEvent.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.event.village;
+
+import net.minecraft.world.entity.npc.Villager;
+import net.minecraft.world.entity.npc.VillagerProfession;
+import net.neoforged.bus.api.ICancellableEvent;
+import net.neoforged.neoforge.common.NeoForge;
+import net.neoforged.neoforge.event.EventHooks;
+import net.neoforged.neoforge.event.entity.EntityEvent;
+
+/**
+ * ChangeProfessionEvent is fired on the server when a Villager Entity changes profession.
+ * This event is fired via the {@link EventHooks#onVillagerProfessionChanged(Villager, VillagerProfession, VillagerProfession)}.
+ * This event is {@link ICancellableEvent cancellable}. If the event is canceled, no profession change will occur.
+ * You can change the profession that the villager will change to via {@link #setNewProfession(VillagerProfession)}.
+ * This event is fired on the {@link NeoForge#EVENT_BUS}.
+ */
+public class VillagerChangeProfessionEvent extends EntityEvent implements ICancellableEvent {
+    private final Villager villager;
```

Comment:
Not sure what the point of this is field is.. the entity is already a field in the super class, so just cast `return (Villager) super.getEntity()`

Reply:
done!